Feat/new password

MiguelDecode opened the pull request:
## Description

Create a page for a user can change the password of the account.
Add HeroLogo and some styles.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (improvement of an existing feature)
- [x] Refactor (code change that neither fixes a bug nor adds a feature)
- [ ] Documentation (changes or additions to documentation)
- [ ] Other (please specify):

## Related Issues

https://trello.com/c/TVPydjGr

## Screenshots

![image](https://github.com/user-attachments/assets/5ffe7324-2b7f-4661-8243-ba47484b0686)

MiguelDecode committed:
feat: add new route to set a new password page

MiguelDecode committed:
feat: basic html structure of set new password page

MiguelDecode committed:
feat: add logical of error message of set a new password

MiguelDecode committed:
fix: add forwardRef to habilitate pass a ref

MiguelDecode commented on src/components/InputPasswordForm/InputPasswordForm.tsx:
I must change a bit this component to habiltate the use of ref as a prop.

Alejandro25AR replied:
If you change the component, you will have to use it with reference everywhere, including registration, login and password recovery, in addition to all operations performed on the profile that request the password.

Alejandro25AR commented on src/pages/NewPasswordPage.tsx:
```diff
@@ -0,0 +1,57 @@
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const password = passwordRef.current?.value;
```
The whole password validation part is handled in the PasswordInputWithRequirements component I am creating. 
![imagen](https://github.com/user-attachments/assets/fddcd1b6-e1ce-40bd-8f72-70dffbb2844b)

MiguelDecode replied:
That's seem more beautiful than I think. Nice work, I'm sure I can't do that.

Alejandro25AR reviewed (commented):
In a moment I will finish the PasswordInputWithRequirements component so that you can use it.

MiguelDecode committed:
feat: remove the useRef logical

MiguelDecode committed:
Revert "fix: remove the useRef and let the componet as original"

This reverts commit 72752e5f8cdd7ae6c7c007af440d4b5c7f167353.

MiguelDecode committed:
Merge branch 'develop' into feat/new-password

MiguelDecode committed:
feat: add securityQuestion component

MiguelDecode committed:
feat: add securityQuestion component

Alejandro25AR committed:
feat: add Checkbox component

Alejandro25AR committed:
feat: add verification icon

Alejandro25AR committed:
feat: add types of TS in folder types

ErrorMessage
Requirements

Alejandro25AR committed:
feat: add reset styles in tag p

Alejandro25AR committed:
feat: add helper helpMessageError

Alejandro25AR committed:
feat: add helper helpValidatePassword

Alejandro25AR committed:
fix: set unique ID in input showPassword button

Alejandro25AR committed:
feat: add component Requirements

This component renders the requirements that the password must fulfill

Alejandro25AR committed:
feat: add component PasswordInputWithRequirements

Is in charge of validating if the added passwords are the same and the
input is valid

MiguelDecode committed:
feat: add InputPhoneForm component

MiguelDecode committed:
fix: change property patter as optional

MiguelDecode committed:
refactor: improve RegisterForm component

MiguelDecode committed:
fix: add fix for no-cors mode

MiguelDecode committed:
fix: pattern problem use default instead

MiguelDecode committed:
feat: add useFetch to send user data

Alejandro25AR committed:
feat: change placeholder

Alejandro25AR committed:
fix: change regular expression in password

MiguelDecode committed:
fix: remove mode no-cors

MiguelDecode committed:
feat: add useFetch to register form

MiguelDecode committed:
feat: add new component PasswordInputWithRequirements

MiguelDecode committed:
fix: modify typo quotes error

MiguelDecode committed:
fix: add status styles

MiguelDecode committed:
fix: modify styles to set the same space between inputs

MiguelDecode committed:
feat: add new page to set a new password

MiguelDecode committed:
Merge branch 'develop' into feat/new-password

MiguelDecode commented:
I'm have a conflict with git and I try to make a rebase but I all the previous commits appear in the history and I think it's awful. Sorry for the commit history.

MiguelDecode committed:
Merge branch 'develop' into feat/new-password

MiguelDecode committed:
fix: add HeroLogo and little styles

Alejandro25AR commented on src/pages/NewPasswordPage.tsx:
```diff
@@ -0,0 +1,53 @@
+const NewPasswordPage = () => {
+  const [password, setPassword] = useState("");
+
+  const url = "http://localhost:8007/user/rpassword/reset";
```
this endpoint is not yet operational

MiguelDecode replied:
Yes, this endpoint not appear in the swagger i take from a trello card.

MiguelDecode committed:
fix: typo error in the endpoint
